### PR TITLE
Feature/92 process 2 need selection

### PR DESCRIPTION
**Pull request by Apemb**

### CODE REVIEW

File: app/javascript/packs/step-2/nextStepButton.vue.erb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+<script>
+    import { mapState, mapMutations, mapActions } from 'vuex'
+    import * as types from './store/mutationTypes'
+
+    export default {
+        name: 'next-step-button',
+        computed: {
+            ...mapState({
+                diagnosisContent: state => state.step2Store.diagnosisContent,
+                isRequestInProgress: state => state.step2Store.isDiagnosisRequestUnderWay
```
Comment (ArnoHolo) [inline app/javascript/packs/step-2/nextStepButton.vue.erb]: Rename UnderWay for `InProgress` or `Outgoing`

File: app/javascript/packs/step-2/nextStepButton.vue.erb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        methods: {
+            nextStepButtonClicked: function () {
+                this.sendDiagnosisContentUdpate()
+                .then( this.createSelectedQuestions() )
+                .then( () => {
+                    Turbolinks.visit('/')
+                });
+            },
+            ...mapMutations({
+                setRequestInProgress: types.DIAGNOSTIC_REQUEST_UNDERWAY
```
Comment (ArnoHolo) [inline app/javascript/packs/step-2/nextStepButton.vue.erb]: `DIAGNOSIS` and `InProgress` or `Outgoing`

File: app/javascript/packs/step-2/nextStepButton.vue.erb (hunk 1 above)
Comment (ArnoHolo) [inline app/javascript/packs/step-2/nextStepButton.vue.erb]: I don't find `isDiagnosisRequestUnderWay` anywhere else in this PR. It is elsewhere in the project?

File: app/javascript/packs/step-2/nextStepButton.vue.erb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        name: 'next-step-button',
+        computed: {
+            ...mapState({
+                diagnosisContent: state => state.step2Store.diagnosisContent,
+                isRequestInProgress: state => state.step2Store.isDiagnosisRequestUnderWay
+            })
+        },
+        methods: {
+            nextStepButtonClicked: function () {
+                this.sendDiagnosisContentUdpate()
```
Comment (ArnoHolo) [inline app/javascript/packs/step-2/nextStepButton.vue.erb]: Update, not Udpate ;) 
(A lot of occurrences in this PR)

File: app/javascript/packs/step-2/appDataSetter.vue.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            }
+        },
+        created: function() {
+            this.setDiagnosisId(this.diagnosisId);
+            this.setDiagnosisContent(this.diagnosisContent);
+        },
+        methods: {
+            ...mapMutations({
+                setDiagnosisId: types.DIAGNOSTIC_ID,
+                setDiagnosisContent: types.DIAGNOSTIC_CONTENT
```
Comment (ArnoHolo) [inline app/javascript/packs/step-2/appDataSetter.vue.erb]: `DIAGNOSIS` 🙏

File: app/javascript/packs/step-2/nextStepButton.vue.erb [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                .then( this.createSelectedQuestions() )
+                .then( () => {
+                    Turbolinks.visit('/')
```
Comment (ArnoHolo) [inline app/javascript/packs/step-2/nextStepButton.vue.erb]: Maybe redirect to somewhere else? (Add a TODO not to forget)

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -34,7 +34,6 @@
   config.before :suite do
     # Run webpack compilation before suite, so assets exists in public/packs
     # see https://github.com/rspec/rspec-core/issues/2366
-    `bin/webpack`
   end
```
Comment (ArnoHolo) [inline spec/rails_helper.rb]: Remove the whole `config.before :suite` block

File: app/javascript/packs/step-2/nextStepButton.vue.erb (hunk 1 above)
Comment (Apemb) [inline app/javascript/packs/step-2/nextStepButton.vue.erb]: renamed `isRequestInProgress`

File: app/javascript/packs/step-2/nextStepButton.vue.erb (hunk 3 above)
Comment (Apemb) [inline app/javascript/packs/step-2/nextStepButton.vue.erb]: Damm

File: app/javascript/packs/step-2/questionSelectionRow.vue.erb [hunk 7]
```diff
@@ -0,0 +1,65 @@
+<template>
+    <tr v-bind:class="{ greyed: selected }">
+      <td>
+        <label v-bind:for="checkboxId">
+          <div class="full-width clickable">
+            {{this.questionLabel}}
```
Comment (ArnoHolo) [inline app/javascript/packs/step-2/questionSelectionRow.vue.erb]: Is `this.` necessary?

File: app/javascript/spec/step-2/store/step2StoreSpec.js [hunk 8]
```diff
@@ -0,0 +1,380 @@
...
+            it('updates the isDiagnosisRequestUnderWay', function () {
+                const state = {isRequestInProgress: false}
+                mutations.REQUEST_IN_PROGRESS(state, true)
+                expect(state.isRequestInProgress).toBeTruthy()
+            })
+        })
+
+        describe('QUESTION_SELECTED', function () {
+
+            it('create the question in the questions array if it does not exist', function () {
```
Comment (ArnoHolo) [inline app/javascript/spec/step-2/store/step2StoreSpec.js]: `creates` (several times in this file)

File: app/javascript/spec/step-2/store/step2StoreSpec.js [hunk 9]
```diff
@@ -0,0 +1,380 @@
...
+                    })
+
+                    expect(commit.calls.count()).toEqual(2)
+                    expect(commit.calls.argsFor(1)).toEqual([
+                        'REQUEST_IN_PROGRESS',
+                        false
+                    ])
+                })
+
+                it('propagtes the error', async function () {
```
Comment (ArnoHolo) [inline app/javascript/spec/step-2/store/step2StoreSpec.js]: `propagates` (several times in this file)

File: config/routes.rb [hunk 10]
```diff
@@ -23,7 +23,11 @@
       patch :update_visitee
     end
 
-    resources :diagnosis, only: %i[index new create show]
+    resources :diagnosis, only: %i[index new create show] do
+      member do
+        get 'step-2' => 'diagnosis#step2'
```
Comment (ArnoHolo) [inline config/routes.rb]: You could remove `member` block and add `, on: :member` at the end of the line, but whatever

File: spec/controllers/api/diagnosed_needs_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+      before do
+        post :create, format: :json, params: { diagnosis_id: diagnosis.id, diagnosed_needs: diagnosed_needs_array }
+      end
+
+      it 'returns http success' do
+        expect(response).to have_http_status(:created)
+      end
+
+      it 'creates two diagnosed needs' do
+        expect(DiagnosedNeed.all.count).to eq(2)
```
Comment (ArnoHolo) [inline spec/controllers/api/diagnosed_needs_controller_spec.rb]: `.all` is not necessary here, but whatever

File: app/javascript/packs/step-2/store/step2APIService.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
+import axios from 'axios'
+
+export default class step2APIService {
+
+    static updateDiagnosisContent(diagnosisId, content) {
+        var config = {
+            method: 'patch',
+            url: `/api/diagnoses/${diagnosisId}`,
```
Comment (ArnoHolo) [inline app/javascript/packs/step-2/store/step2APIService.js]: Generate this path from Rails

File: app/javascript/packs/step-2/store/step2APIService.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        return this.send(config).then(() => {
+            return true
+        })
+    }
+
+    static createDiagnosedNeeds(diagnosisId, diagnosedNeeds) {
+        var config = {
+            method: 'post',
+            url: `/api/diagnoses/${diagnosisId}/diagnosed_needs`,
```
Comment (ArnoHolo) [inline app/javascript/packs/step-2/store/step2APIService.js]: Generate this path from Rails

**Pull request merged**